build-tools/packages/build-tools/src/fluidBuild/buildGraph.ts

```diff
@@ -214,13 +234,13 @@ export class BuildPackage {
 			if (task.taskName === undefined) {
 				return;
 			}
-			const taskConfig = this.taskDefinitions[task.taskName];
+			const taskConfig = this.getTaskDefinition(task.taskName);
 			if (taskConfig === undefined) {
 				return;
 			}
 			if (taskConfig.before.includes("*")) {
 				this.tasks.forEach((depTask) => {
-					if (depTask !== task) {
+					if (depTask !== task && !task.dependentTasks!.includes(depTask)) {
```

Comment:
is dependentTasks guaranteed to be defined?

Reply:
At this point in time, `initializeDependentTask` should be called on all the task already.  Added comment.